game-server/data/handlers/quest/theobomos/_3058StoneofMabolo.java

```diff
@@ -83,15 +84,30 @@ else if (env.getDialogActionId() == SELECT_QUEST_REWARD)
 					return false;
 				case SETPRO2:
 					if (var == 1) {
```

Comment:
You don't need to check the `var` here anymore, since you're using `defaultCloseDialog(env, 1, 2)` which checks the quest var internally.
This simplifies the whole `SETPRO2` case to a single return statement.